src/hooks/useGasPrice.ts

```diff
@@ -53,10 +96,35 @@ const getGasPrice = async (gasPriceConfigs: GasPrice): Promise<BigNumber | undef
   }
 }
 
-const useGasPrice = (): AsyncResult<{
-  maxFeePerGas: BigNumber | undefined
-  maxPriorityFeePerGas: BigNumber | undefined
-}> => {
+const getGasParameters = (
+  estimation: EstimatedGasPrice | undefined,
+  feeData: FeeData | undefined,
+  isEIP1559: boolean,
+): GasFeeParams => {
+  if (!estimation) {
+    return {
+      maxFeePerGas: isEIP1559 ? feeData?.maxFeePerGas : feeData?.gasPrice,
+      maxPriorityFeePerGas: isEIP1559 ? feeData?.maxPriorityFeePerGas : undefined,
+    }
+  }
+
+  if (isEIP1559 && 'maxFeePerGas' in estimation && 'maxPriorityFeePerGas' in estimation) {
+    return estimation
+  }
+
+  if ('gasPrice' in estimation) {
+    return {
+      maxFeePerGas: estimation.gasPrice,
+      maxPriorityFeePerGas: isEIP1559 ? feeData?.maxPriorityFeePerGas : undefined,
+    }
+  }
+
+  return {
+    maxFeePerGas: undefined,
+    maxPriorityFeePerGas: undefined,
+  }
+}
```

Comment:
This looks more readable now but I would suggest to write a few tests to cover some of the edge-cases like if there is no estimation and it is EIP1559 but there is also no feeData etc.